gridcapa-core-cc-app/src/main/java/com/farao_community/farao/gridcapa_core_cc/app/domain/CoreCCTaskParameters.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ *  This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.farao_community.farao.gridcapa_core_cc.app.domain;
+
+import com.farao_community.farao.gridcapa.task_manager.api.TaskParameterDto;
+import com.farao_community.farao.gridcapa_core_cc.api.exception.CoreCCInvalidDataException;
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * @author Daniel Thirion {@literal <daniel.thirion at rte-france.com>}
+ */
+public class CoreCCTaskParameters {
+    private static final Logger LOGGER = LoggerFactory.getLogger(CoreCCTaskParameters.class);
+    private static final String USE_DC_CGM_INPUT = "USE_DC_CGM_INPUT";
+    private static final String KEY_VALUE_FORMAT = "%n\t\"%s\": %s";
+    private boolean useDcCGmInput;
+
+    public CoreCCTaskParameters(final List<TaskParameterDto> parameters) {
+        final List<String> errors = new ArrayList<>();
+        for (final TaskParameterDto parameter : parameters) {
+            if (USE_DC_CGM_INPUT.equals(parameter.getId())) {
+                useDcCGmInput = validateIsBooleanAndGet(parameter, errors);
+            } else {
+                LOGGER.warn("Unknown parameter {} (value: {}) will be ignored", parameter.getId(), parameter.getValue());
+            }
+        }
+        if (!errors.isEmpty()) {
+            final String message = String.format("Validation of parameters failed. Failure reasons are: [\"%s\"].", String.join("\" ; \"", errors));
+            throw new CoreCCInvalidDataException(message);
+        }
+    }
+
+    private boolean validateIsBooleanAndGet(final TaskParameterDto parameter,
+                                            final List<String> errors) {
+        if (StringUtils.equals("BOOLEAN", parameter.getParameterType())) {
+            final String value = parameter.getValue() != null ? parameter.getValue() : parameter.getDefaultValue();
+            return Boolean.parseBoolean(value);
+        } else {
+            errors.add(String.format("Parameter %s was expected to be of type BOOLEAN, got %s", parameter.getId(), parameter.getParameterType()));
+            return false;
+        }
+    }
+
+    public boolean isUseDcCGmInput() {
+        return useDcCGmInput;
+    }
+
+    public String toJsonString() {
```

Comment:
I don't know what you want to do exactly here, but maybe this could do the same?
```java
ToStringBuilder.reflectionToString(this, ToStringStyle.JSON_STYLE);
```

Do you need the parameter name to be in snake_case and capital letters?

Reply:
As discussed in real life today, I used same parameter-logging mechanics that in SweTaskParameters.java, even though i agree it makes less sense here since only one parameter is present